### PR TITLE
[sk] Some cleaning rule bug fixes

### PR DESCRIPTION
**Pull request by skunichetty**
# Summary
Fixed the following bugs:
- Floating point values now no longer get truncated to integers
- Clean column name now strips whitespaces and surrounding dashes from title
- Datetime parsing now automatically handles out of bounds dates
- Datetime regex updated to prevent accidental parsing of ids like xxx-xx-xxxx

# Tests
Unit tests were performed locally and tests on external datasets were also performed.

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_cleaner/shared/utils.py [hunk 1]
```diff
@@ -12,14 +8,13 @@
 def clean_series(series, column_type, dropna=True):
     series_cleaned = series.apply(lambda x: x.strip(' \'\"') if type(x) is str else x)
     series_cleaned = series_cleaned.map(
-        lambda x: x if (not isinstance(x, str) or (len(x) > 0 and not x.isspace())) else np.nan,
-    )
+        lambda x: x if (not isinstance(x, str) or (len(x) > 0 and not x.isspace())) else np.nan,)
```
Comment (wangxiaoyou1993) [inline mage_ai/data_cleaner/shared/utils.py:11]: nit: remove the comma after np.nan

File: mage_ai/tests/data_cleaner/statistics/test_calculator.py [hunk 2]
```diff
@@ -23,12 +22,13 @@ def test_calculate_statistics_overview(self):
             [1, None, '2000-07-01 00:00:00+00:00', '5', False],
             ['', None, '899-07-01 00:00:00+00:00', '6', False],
             [3, 'US', None, '7', True],
-        ], columns=['age', 'country', 'date_joined', 'id', 'cancelled'])
+        ],
+                          columns=['age', 'country', 'date_joined', 'id', 'cancelled'])
```
Comment (wangxiaoyou1993) [inline mage_ai/tests/data_cleaner/statistics/test_calculator.py:26]: the style is weird

Comment (skunichetty) [reply mage_ai/tests/data_cleaner/statistics/test_calculator.py]: Not sure how this happened, but resolved now.

File: mage_ai/data_cleaner/shared/utils.py (hunk 1 above)
Comment (skunichetty) [reply mage_ai/data_cleaner/shared/utils.py]: Resolved

**Comment by wangxiaoyou1993** [comment]
Could you rebase master? There're some duplicate code with your previous PR.

**Review by wangxiaoyou1993 — APPROVED** [review]

File: mage_ai/data_cleaner/pipelines/base.py [hunk 3]
```diff
@@ -52,16 +51,16 @@ def transform(self, df, auto=True):
         if len(self.actions) == 0:
             print('Pipeline is empty.')
             return df
-        action_queue = deque(self.actions)
+        action_queue = self.actions
         completed_queue = []
         df_transformed = df
         while len(action_queue) != 0:
-            action = action_queue.popleft()
+            action = action_queue[0]
```
Comment (wangxiaoyou1993) [inline mage_ai/data_cleaner/pipelines/base.py:58]: @skunichetty This actually causes infinite loop when auto is False.

**Pull request merged**